python/datasources/uhc.py

```diff
@@ -145,150 +140,208 @@ def upload_to_gcs(self, _, **attrs):
+       state: string state name"""
+
+    pres_breakdown_value, mid_breakdown_value = np.nan, np.nan
 
-                            pres_breakdown_value, mid_breakdown_value = np.nan, np.nan
+    if len(matched_row) > 0:
+        pres_breakdown_value = matched_row['Value'].values[0]
 
-                            if len(matched_row) > 0:
-                                pres_breakdown_value = matched_row['Value'].values[0]
+    matched_row_midterm = df.loc[
+        (df[std_col.STATE_NAME_COL] == state) &
+        (df['Measure Name'] == measure_name)]
 
-                            matched_row_midterm = df.loc[
-                                (df['State Name'] == state) &
-                                (df['Measure Name'] == measure_name)]
+    if len(matched_row_midterm) > 0:
+        mid_breakdown_value = matched_row_midterm['Value'].values[0]
 
-                            if len(matched_row_midterm) > 0:
-                                mid_breakdown_value = matched_row_midterm['Value'].values[0]
+    average_value = np.nanmean(
+        [pres_breakdown_value, mid_breakdown_value])
 
-                            average_value = np.nanmean(
-                                [pres_breakdown_value, mid_breakdown_value])
+    return average_value * 1000
 
-                            output_row[std_col.VOTER_PARTICIPATION_PER_100K] = average_value * 1000
 
-                        # for other determinants besides VOTER
-                        elif len(matched_row) > 0:
-                            pct = matched_row['Value'].values[0]
-                            if pct:
-                                if determinant in PER100K_DETERMINANTS:
-                                    output_row[UHC_DETERMINANTS[determinant]
-                                               ] = matched_row['Value'].values[0]
-                                # convert from % to per 100k
-                                else:
-                                    output_row[UHC_DETERMINANTS[determinant]
-                                               ] = matched_row['Value'].values[0] * 1000
+def estimate_total(row, condition_name_per_100k):
+    """Returns an estimate of the total number of people with a given condition.
 
-                output.append(output_row)
+       sample_per_100k: a percentage of people in a demographic with a given condition, represented as a per 100k
+       total_population: the total number of people in that demographic"""
 
-        output_df = pd.DataFrame(output, columns=columns)
+    if pd.isna(row[condition_name_per_100k]) or \
+        pd.isna(row[std_col.POPULATION_COL]) or \
+            int(row[std_col.POPULATION_COL]) == 0:
 
-        if breakdown == std_col.RACE_OR_HISPANIC_COL:
-            std_col.add_race_columns_from_category_id(output_df)
+        return None
 
-        return output_df
+    return round((float(row[condition_name_per_100k]) / 100000) * float(row[std_col.POPULATION_COL]))
```

Comment:
should `condition_name_per_100k` be something like `condition_raw_count` instead? I'm not clear why you need to divide a per100k rate by 100k again.

Reply:
you divide the `per_100k` number to get the raw ratio, which you then multiply by the population.